app/src/cobrand/java/cm/aptoide/pt/billing/view/login/PaymentLoginFlavorPresenter.java

```diff
@@ -0,0 +1,142 @@
+package cm.aptoide.pt.billing.view.login;
```

Comment:
Personally not a big fan of inheritance both here and on LoginSignupCredentialsPresenter for two reasons:

1) Conceptually there's no real reason to have inheritance in presenters other than repeated code because of different flavours. If we have X flavours, we will need X children. Even though unlikely to happen, it might present a scalability problem if we were to create other flavours.
2) It ironically leads to repeated code, check "handleAptoideSignUpEvent()" for example. The difference is two lines, the rest is repeated code. Again, if we need a new flavour we will have to repeat this because we likely will need to sign up.

Personally I prefer to isolate the differences as much as possible and avoid repeated code. How about having like a "PolicyManager" where it would hold whether it should show PP and TC and other related behaviour? Then you would only have one presenter that would react accordingly if we should show or not. This also gives flexibility if we want to change the visibility, for some reason, on the vanilla flavour.

Reply:
Removed unnecessary code, and extracted methods to prevent reuse of code. Also, opened a ticket for the future, for these classes to be refactored and remove the use of inheritance.